### PR TITLE
Add Apache 2.0 license

### PR DESCRIPTION
**Pull request by kmclarnon**
@jhaber @stevegutz

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (aaylward) [inline LICENSE.txt]: is this right?

Comment (kmclarnon) [reply LICENSE.txt]: Definitely not, copypasta failure on my part

Comment (kmclarnon) [reply LICENSE.txt]: Seems to be common https://github.com/HubSpot/dropwizard-guice/blob/master/LICENSE.txt#L189 😢

Comment (kmclarnon) [reply LICENSE.txt]: Fixed, thanks @aaylward 👍

Comment (jhaber) [reply LICENSE.txt]: Github has a feature that lets you add a license from a predefined list, not sure if it handles the year for you

Comment (kmclarnon) [reply LICENSE.txt]: This file was added via github's build-in functionality, seems like it doesn't handle either the year or the org

Comment (jhaber) [reply LICENSE.txt]: Then all the ones I've added are probably messed up 😂

**Pull request merged**